models/user.py

```diff
@@ -1,38 +1,85 @@
-from models.BaseModel import BaseModel
+from utility.helpers import query
+from utility.decor import path_set
 
 
-class User(BaseModel):
-
+####
+class User():
     def __init__(self):
-        self.email_userid = {}
-        super().__init__()
+        pass
+
+    @path_set(path='SQL/user')
+    def add(self, client_name, email, password, path):
 
-    def add(self, entry):
-        userid = super().add(entry)
-        email = entry['email']
-        self.email_userid[email] = userid
+        query(path, 'register',
+              vals=(client_name, email, password))
 
-    def validate(self, email, password):
+    @path_set(path='SQL/user')
+    def validate(self, email, password, path):
 
-        if self.old_user(email):
-            userid = self.email_userid[email]
-            entry = self.find_val(userid)
-            return entry['password'] == password
+        rows = query(path, 'validate', vals=(email,))
+        # print(rows)
+        if rows is None:
+            return False
+        elif rows == []:
+            return False
         else:
-            False
+            inquired_password = rows[0]['passkey']
+            return password == inquired_password
+
+    @path_set(path='SQL/user')
+    def old_user(self, email, path):
+
+        email_list = query(path, 'old_user')
+        email_list = [x['email'] for x in email_list]
+        return email in email_list
```

Comment:
Sometimes, it might be better to be more verbose. Also, note how these two pieces of code behave differently when the number of records is large (which is not the case here).

```suggestion
        for record in email_list:
            if email == record['email']:
                return True
        return False
```